libcraft/core/src/dimension.rs

```diff
@@ -0,0 +1,68 @@
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "&'static str")]
+pub enum Dimension {
+    Overworld,
+    TheNether,
+    TheEnd,
+}
+
+impl Dimension {
+    pub fn dim_id(&self) -> i32 {
```

Comment:
Should maybe implement `From<i32> for Dimession` and `From<Dimension> for i32` in addition.

Reply:
I think it should be `TryFrom<i32> for Dimension` as the conversion can fail.

Reply:
Ahh yes, indeed it should be.

Reply:
Also this is merely a suggestion, I'm not sure if we should implement `From` or `TryFrom<i32>` I'm leaning towards not, just wanted to bring up the possibility.

Reply:
Well, he has implemented it already so we will have the possibility xD.